packages/clarity-js/src/performance/interaction.ts

```diff
@@ -0,0 +1,124 @@
+// Map to track interactions by their ID, ensuring we handle duplicates
+const longestInteractionMap: Map<number, Interaction> = new Map();
+
+/**
+ * Update the approx number of interactions estimate count if the interactionCount is not supported.
+ * The difference between `maxKnownInteractionId` and `minKnownInteractionId` gives us a rough range of how many interactions have occurred. 
+ * Dividing by 7 helps approximate the interaction count more accurately, since interaction IDs may not always increase sequentially.
+ */
+const countInteractions = (entry: PerformanceEventTiming) => {
+  if ('interactionCount' in Performance) return;
+
+  if (entry.interactionId) {
+    minKnownInteractionId = Math.min(
+      minKnownInteractionId,
+      entry.interactionId
+    );
+    maxKnownInteractionId = Math.max(
+      maxKnownInteractionId,
+      entry.interactionId
+    );
+
+
+    interactionCountEstimate = maxKnownInteractionId
+      ? (maxKnownInteractionId - minKnownInteractionId) / 7 + 1
+      : 0;
+  }
+};
+
+const getInteractionCount = () => {
+  return (
+    'interactionCount' in performance
+      ? performance.interactionCount
+      : interactionCountEstimate || 0
+  ) as number;
+};
+
+const getInteractionCountForNavigation = () => {
+  return getInteractionCount() - prevInteractionCount;
+};
+
+/**
+ * Estimates the 98th percentile (P98) of the longest interactions by selecting
+ * the candidate interaction based on the current interaction count.
+ */
+export const estimateP98LongestInteraction = () => {
+  const candidateInteractionIndex = Math.min(
+    longestInteractionList.length - 1,
+    Math.floor(getInteractionCountForNavigation() / 50)
+  );
+
+  return longestInteractionList[candidateInteractionIndex].latency;
+};
+
+/**
+ * Resets the interaction tracking, usually called after navigation to a new page.
+ */
+export const resetInteractions = () => {
+  prevInteractionCount = getInteractionCount();
+  longestInteractionList.length = 0;
+  longestInteractionMap.clear();
+};
+
+/**
+ * Processes a PerformanceEventTiming entry by updating the longest interaction list.
+ *
+ * @param entry - A new PerformanceEventTiming entry for an interaction
+ */
+export const processInteractionEntry = (entry: PerformanceEventTiming) => {
+  // Ignore entries with 0 interactionId or very short durations
+  if (!entry.interactionId || entry.duration < DEFAULT_DURATION_THRESHOLD) {
+    return;
+  }
+
+  // Count the interactions if manual counting is necessary
+  countInteractions(entry);
+
+  const minLongestInteraction =
+    longestInteractionList[longestInteractionList.length - 1];
+
+  const existingInteraction = longestInteractionMap.get(entry.interactionId!);
+
+  // Either update existing, add new, or replace shortest interaction if necessary
+  if (
+    existingInteraction ||
+    longestInteractionList.length < MAX_INTERACTIONS_TO_CONSIDER ||
+    entry.duration > minLongestInteraction.latency
```

Comment:
>minLongestInteraction [](http://example.com/codeflow?start=21&length=21)

don't you run the risk of a null reference here? On our first execution longestInteractionList was [] so minLongestInteraction will be undefined, which should throw at this point?